realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -2065,7 +2065,9 @@ private OsResults lazyFindAll() {
      * subscription by default.
      * <p>
      * This predicate can be called multiple times, in which case all fields will be added to the subscription.
-     * 
+     * <p>
+     * NOTE: This method version is only available with Realm Object Server 3.21.1 or later.
```

Comment:
```suggestion
     * NOTE: This method is only supported when connecting to Realm Object Server 3.21.0 or later. If you use it with previous versions of Realm Object Server, an 'InvalidSyntaxException' will be observable via the ChangeListeners getError() method on the changeset returned.
```
Perhaps rephrase the error returned?